modules/data.land/R/soilgrids_soc_extraction.R

```diff
@@ -66,6 +66,18 @@ soilgrids_soilC_extract <- function (site_info, outdir=NULL, verbose=TRUE) {
   
   # prepare site info for extraction
   internal_site_info <- site_info[, c("site_id", "site_name", "lat", "lon")]
+
+  # Early return if no valid sites (after processing internal_site_info)
+  if (nrow(internal_site_info) == 0) {
+    if (verbose) {
+      PEcAn.logger::logger.severe(
+        "No valid sites remaining after NA check. ",
+        "All sites had missing SoilGrids data for the first depth layer."
+      )
+    }
```

Comment:
1. Message seems out of place -- no NA check has happened yet at this point. for nrows to be zero it has to be passed that way by the user, so it could definitely happen for any number of reasons that aren't the one the message says. 
2. anything bad enough to warrant logger.severe (which will stop execution of the entire workflow) should not be suppressible by the verbose flag.

```suggestion
    if (verbose) {
      PEcAn.logger::logger.info("No sites specified")
    }
```